### PR TITLE
Add ASK class

### PR DESCRIPTION
**Pull request by kuro11pow2**
# Major Change
## Wave
* add mAmpSize to adjust the magnitude of the wave
* add setter and getter for mAmpSize
## WaveBuilder
* add setAmpSize method
## ModemFactory
* add modulation type

# Description
For a string composed of 0 and 1, modulate/demodulate is performed with ASK.
This implementation may be inappropriate. We hope for a careful review of this.

Co-authored-by: jopopcorn <jopopcorn@users.noreply.github.com>
Co-authored-by: psh2849 <psh2849@users.noreply.github.com>
Co-authored-by: HACore <HACore@users.noreply.github.com>

### CODE REVIEW

File: euphony/src/main/cpp/core/source/ASK.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        }
+    }
+
+    return result;
+}
+
+shared_ptr<Packet> ASK::demodulate(const WaveList& waveList) {
+    HexVector hexVector = HexVector(waveList.size());
+    const float threshold = 0.009;
+    const int startIdx = getStartFreqIdx(); // fsk에서 사용되는 것과 동일
```
Comment (psh2849) [inline euphony/src/main/cpp/core/source/ASK.cpp:53]: It would be better to delete comments!

**Review by HACore — COMMENTED** [review]
built successfully in my environment too!
GOOD👍👍👍

File: euphony/src/main/cpp/core/source/ASK.cpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+shared_ptr<Packet> ASK::demodulate(const WaveList& waveList) {
+    HexVector hexVector = HexVector(waveList.size());
+    const float threshold = 0.009;
+    const int startIdx = getStartFreqIdx();
+
+    for(const auto& wave : waveList) {
+        auto vectorInt16Source = wave->getInt16Source();
+        int16_t* int16Source = &vectorInt16Source[0];
+        float *resultBuf = fftModel->makeSpectrum(int16Source);
```
Comment (designe) [inline euphony/src/main/cpp/core/source/ASK.cpp:58]:
```suggestion
        auto spectrums = fftModel->makeSpectrum(int16Source);
        float *resultBuf = spectrums.amplitudeSpectrum;
```

Comment (designe) [reply euphony/src/main/cpp/core/source/ASK.cpp:58]: `fftModel` interface has been changed now. let me fix & apply here :)

**Review by designe — APPROVED** [review]
I think It's really great implementation of `ASK` :+1: 
Let me merge with `master` as soon as possible :)

**Pull request merged**